packages/datadog-instrumentations/src/check_require_cache.js

```diff
@@ -39,3 +61,35 @@ module.exports = function () {
 
   if (didWarn) console.error('Warning: Please ensure dd-trace is loaded before other modules.')
 }
+
+/**
+ * APM tools, and some other packages in the community, work
+ * by monkey-patching internal modules and possibly some
+ * globals. Usually this is done in a conflict-free way by
+ * wrapping an existing method with a new method that still
+ * calls the original method. Unfortunately it's possible
+ * that some of these packages (dd-trace included) may
+ * wrap methods in a way that make it unsafe for the methods
+ * to be wrapped again by another library.
+ *
+ * When encountered, and when debug mode is on, a warning is
+ * printed if such a package is discovered. This can help
+ * when debugging a faulty installation.
+ */
+module.exports.checkForPotentialConflicts = function () {
+  const naughties = new Set()
+  let didWarn = false
+
+  for (const pathToModule of Object.keys(require.cache)) {
+    const { pkg } = extractPackageAndModulePath(pathToModule)
+    if (naughties.has(pkg)) continue
+    if (!potentialConflicts.has(pkg)) continue
+
+    console.error(`Warning: Package '${pkg}' may cause conflicts with dd-trace.`)
```

Comment:
Should this and the next one be either `process.emitWarning()` or `console.warn()`? I'm not sure.

Reply:
I changed this to `console.warn`.

For some reason using `process.emitWarning` at that location will not print any output. I grepped the codebase and node_modules and can't figure out why. Printing the stringified method suggests it's the one built into node. Running `node -e "process.emitWarning('foo')"` works fine too. 🤷🏻

Reply:
This was further changed to queue the messages and then log once the tracer has initialized.